src/app/editor-common/components/key-value-component/key-value-input.component.ts

```diff
@@ -0,0 +1,129 @@
+import {Component, forwardRef, Input} from "@angular/core";
+import {
+    NG_VALUE_ACCESSOR,
+    ControlValueAccessor,
+    FormGroup,
+    FormBuilder,
+    NG_VALIDATORS,
+    FormControl
+} from "@angular/forms";
+import {ExpressionModel} from "cwlts/models/d2sb";
+import {ComponentBase} from "../../../components/common/component-base";
+
+require("./key-value-input.component.scss");
+
+@Component({
+    selector: "[ct-key-value-input]",
```

Comment:
Though I don't think it's explicitly wrong to have an attribute selector for a component, the trend so far in the application has been to keep attribute selectors to directives. Any reason why this needs to be an attribute selector?